### ISSUE TITLE
Playervault open item

### ISSUE THREAD

**Issue by Flavourized** [drtshock/PlayerVaults#316]
This feature adds the option to open the PVX

Some servers (like mine) have playervault set up in a way where they can only be accessed through the PV sign. This means that the player has to go to the designated area, where the PV signs are, to open a vault.

 This feature will allow players to open their PV with this item regardless of where they are. PVX will give the player this item* which they will hold in their hand. The player will then right click with the item in hand. PVX will then remove the item* and then open the PV GUI, where the player can pick which vault to open.

I don't think I need to elaborate on the GUI; from the discord server, I think we all have a good idea how the GUI will function.

*ITEM
This item should have settings in config.yml. i would like the ability to set:
- material_name or item_name 
- item_data
- name of the item
- lore lines
- Endless item**
- Enchant on the item
- message when item is successfully consumed
- a cooldown time between using this item 2 times in a row
- Region/World blacklist to deny using this item

** Endless item means there is a version of this item which has unlimited uses; the item should remain in the inventory even after the player has used it. This should not be a default option. 

There should be an argument where the server owner can specify if they want the item to be endless or not.

I know you are asking yourself "Why add endles version?? It is the damn thing as using the command". So, with the idea of opening PVs using signs only, having this item would be somewhat rare, and people should be able to get it if another player dies, or drops it, etc. 

-----

Settings example:

PVItem:
  Material: 'stick'
  Data: 0
  Enchants:
    - '34:10' # unbreaking 10
    - 'silk_touch:1' #idk why someone would use 2 enchants but the option should be there anyways
  CustomName: "&c&lplayervault express"
  Lore:
    - 'lore line 1'
    - 'lore line X'
  onOpenMessage: "(cool server) you opened the vaults"
  onCloseMessage: "(hot server) you close the vaults"

-----
Commands example, <required> [optional]

/pv giveitem <player> <amount> <Endless? (True/false>
  > /pv giveitem Flavour 1 true - gives endless item
  > /pv giveitem Flavour 1 false - gives consumable
  
--
I know i probably don't need to say it, but for server with 99 vaults, there would need to be pages in the GUI. Pages means there will need to be buttons like Next, Last, close, etc. I wanted to say that these should customizable so we can manipulate the GUI how we want it. Basically let us choose what buttons go where (or if we want to disable a button type)

**Comment by Flavourized** [comment]
Also forgot mention adding permissions to use these!
 
playervaults.openitem.use
playervaults.openitem.endless.use
playervault.openitem.give

1) permission needed for player to use an item. If they don't have this item, return a no perms msg
2) same thing except for endless version if the item
3) item give - for admins

**Comment by drtshock** [comment]
Thank you for the detail! Anyone else want to comment on this implementation?

**Comment by TWSSYesterday** [comment]
IMHO this should be an addon because currently the plugin is good because it doesn't have these features. Lots of servers wouldn't use this, mine included and I feel that having it as a module/addon would be much better.

**Comment by Flavourized** [comment]
>  having it as a module/addon would be much better.

not trying to be mean or anything, but what difference would it make if it was a separate jar, or baked into the 1 jar. If you're not going to use it (and no part of your server uses it), then it wouldnt take resources, unless im mistaken? Isn't this what lazy loading is ? :L

Baking it into 1 jar allows easier updating whenever there are major changes, and simply put: convenience

**Comment by drtshock** [comment]
Yeah anything like this wouldn't be loaded if you wanted it disabled. I would just have a config option to make it like it doesn't exist, which is what it would default to.

**Comment by Flavourized** [comment]
Lax implemented this feature into his PlayerVaults I plugin. see here: https://github.com/AwesomePowered/PlayerVaultsGUI/issues/1
